app/src/main/java/net/squanchy/schedule/ScheduleService.kt

```diff
@@ -32,13 +33,15 @@ class FirestoreScheduleService(
 ) : ScheduleService {
 
     override fun schedule(onlyFavorites: Boolean): Observable<Schedule> {
-        return dbService.scheduleView()
-            .map { schedulePages ->
+        return Observable.combineLatest(dbService.scheduleView(), dbService.timezone(), combineInAPair())
+            .map { pagesAndTimeZone ->
+                val schedulePages = pagesAndTimeZone.first
+                val timeZone = pagesAndTimeZone.second
```

Comment:
not complitely sure but you might use a destructuring declaration, like:
`val (schedulePages, timeZone) = pagesAndTimeZone`